charts/authz/templates/deployment.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.deployment.spec.replicas }}
+  selector:
+    matchLabels:
+      {{- include "labels" . | indent 6 }}
+  strategy:
+    type: {{ .Values.deployment.strategy.type }}
+    rollingUpdate:
+      maxSurge: {{ .Values.deployment.strategy.rollingUpdate.maxSurge }}
+  template:
+    metadata:
+      labels:
+        {{- include "labels" . | indent 8 }}
+    spec:
+      serviceAccountName: {{ .Values.name }}-service-account
+      automountServiceAccountToken: true
+      containers:
+      - name: nginx
+        image: nginx:{{ .Values.deployment.nginx.tag }}
+        ports:
+        - containerPort: 80
+          name: http
+          protocol: TCP
+        volumeMounts:
+          - name: nginx-config
+            mountPath: /etc/nginx
+      - name: {{ .Values.name }}
```

Comment:
Can we also add env vars support so we can configure our app

Reply:
please add liveness and readiness probe to the container

Reply:
Added env vars support and liveness and readiness probe to the container